### ISSUE TITLE
Resolving `electron` is attempted if listed in `devDependencies` but not installed (due to `--production`)

### ISSUE THREAD

**Issue by simotukiainen** [electron/electron-packager#1219]
Dear `electron-packager` people, thanks for working on this great package.

### Preflight Checklist

* [x] I have read the [contribution documentation](https://github.com/electron/electron-packager/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [code of conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project follows, as appropriate.
* [x] I have searched the issue tracker for a bug that matches the one I want to file, without success.

### Issue Details

* **Electron Packager Version:**
  * 15.2.0
* **Electron Version:**
  * 11.2.2
* **Operating System:**
  * Node 14.15.4, Linux
* **Last Known Working Electron Packager version:**:
  * N/A

### Expected Behavior
When building in a directory with no `devDependencies` installed and `electron` listed in `devDependencies`, build should not fail due to not being able to resolve `electron` listed in `devDependencies`

### Actual Behavior
The build fails with:
```
Failed to locate module "electron" from "."
```

### To Reproduce

Use globally or elsewhere installed `electron-packager` and this `package.json` in an empty directory:
```
{
  "name": "eptest",
  "version": "0.0.0",
  "description": "",
  "dependencies": {
    "lodash": "^4.17.20"
  },
  "devDependencies": {
    "electron": "^11.2.2"
  },
  "author": "",
  "license": "Apache-2.0"
}
```
(The only purpose of `lodash` is to have at least one real dependency.)

Then install with `npm ci --production` to not to install `devDependencies`.

And finally try to package: `electron-packager --electronVersion 11.2.2 .`. Note that we are giving `electronVersion` so that it doesn't need to resolve `electron` to determine its version.
This, however, results in:
```
Failed to locate module "electron" from "."
```
which I suspect is related to `electron-packager` attempting to resolve the `devDependenciecs` module to ignore it.

Workarounds:
- If `devDependencies` are removed from `package.json`, it works!
- If there are no production packages, it also works.
- If `electron` is found in a super directory, it works. Actually this is worrying, because it potentially means other directories affect my clean build which I'm especially trying to avoid by copying only necessary files to a clean directory (it's under build/ so that's why project directory packages get looked at—perhaps I will copy to `/tmp` to avoid this). It doesn't attempt to use global `electron` though.

### Additional Information

So basically what I'm doing is that I have a build script that copies only files I want to distribute into a clean directory along with `package.json` and then does `npm ci --production` over there before packaging it with `electron-packager` from a separate directory (my build scripts has its own `package.json` to separate packaging tools from the app). I was happy to find out that `electron-packager` does not in fact need `electron` installed if `--electronVersion` is used. I worked around this by having my build script delete the `devDependencies` from clean `package.json` before building and it works, so I guess it's not a very big problem since most people build with `electron` etc. as `devDependencies` and installed.

**Comment by malept** [comment]
Your steps to reproduce have some issues with it, including that there's no `main` key in your package.json. I suspect that may be the error you're getting. I can't reproduce the behavior you describe with the following steps (on Linux/x64):

```
git clone https://github.com/electron/electron-quick-start
cd electron-quick-start
npm ci --production
npx --no-install --ignore-existing electron-packager . --electron-version=11.2.2
# run the packaged app
electron-quick-start-linux-x64/electron-quick-start
```

**Comment by simotukiainen** [comment]
Thanks for checking this out.
I'm pretty sure `main` key doesn't have anything to do with this. Also I should had been more clear about it but the problem happens during build and running has nothing to do with it.

As I mentioned, the build works if there are no production dependencies (and `electron-quick-start` has no production deps). Thanks for showing this easier way to reproduce it—if I add one production package in your example, it fails:
```
git clone https://github.com/electron/electron-quick-start
cd electron-quick-start
npm ci --production
npm install --production lodash
npx --no-install --ignore-existing electron-packager . --electron-version=11.2.2
```
results in:
```
Packaging app for platform linux x64 using electron v11.2.2
Failed to locate module "electron" from "."

        This normally means that either you have deleted this package already somehow (check your ignore settings if using electron-packager).  Or your module installation failed.
```

Would you mind trying it again with that added step?

Here is also my DEBUG=* output with this example:
https://gist.github.com/c183eeb10edc74b4d1d85dc3959509b3

**Comment by malept** [comment]
I'm going to mark this as "won't fix", because Flora Colossus, the node module walker that Electron Packager uses, assumes that you haven't just installed the production dependencies. Packager prunes those for you anyway.

If you want to continue using this method, specify `--no-prune`.
